### PR TITLE
[catalog] Update api documentation

### PR DESCRIPTION
**Pull request by alde**
Update catalog api documentation for location endpoint.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by sennyeya** [comment]
@alde These are great docs! Can the descriptions and status codes be added to the catalog OpenAPI spec 🙏 ? I'm happy to update as well.

**Review by vinzscam — APPROVED** [review]

File: docs/features/software-catalog/api.md [hunk 1]
```diff
@@ -279,7 +279,63 @@ where the `items` array has _the same length_ and _the same order_ as the input
...
+Response type is JSON, on the form
+
+```json
+{
+  "data": {
+    "id": "b9784c38-7118-472f-9e22-5638fc73bab0",
+    "target": "https://git.example.com/example-project/example-repository/blob/main/catalog-info.yaml",
+    "type": "url"
+  }
+}
```
Comment (Rugvip) [inline docs/features/software-catalog/api.md:295]:
```suggestion
[
  "data": {
    "id": "b9784c38-7118-472f-9e22-5638fc73bab0",
    "target": "https://git.example.com/example-project/example-repository/blob/main/catalog-info.yaml",
    "type": "url"
  }
]
```

Right? 😅

**Review by Rugvip — APPROVED** [review]
Nice 😁

File: docs/features/software-catalog/api.md (hunk 1 above)
Comment (alde) [reply docs/features/software-catalog/api.md:295]: Actually no, both are incorrect. Will fix.

**Pull request merged**